src/details/ArborX_DetailsTreeTraversal.hpp

```diff
@@ -82,7 +76,7 @@ struct TreeTraversal<BVH, Predicates, Callback, SpatialPredicateTag>
   KOKKOS_FUNCTION void operator()(OneLeafTree, int queryIndex) const
   {
     auto const &predicate = Access::get(_predicates, queryIndex);
-    Node const *root = HappyTreeFriends::getRoot(_bvh);
+    auto const root = HappyTreeFriends::getRoot(_bvh);
```

Comment:
Please use `root_index` instead of `root` in all places where it is an `int`. I get really confused otherwise. And it follows `node_index` style.